src/detect-engine.c

```diff
@@ -4938,6 +4952,35 @@ void DetectEngineSetEvent(DetectEngineThreadCtx *det_ctx, uint8_t e)
     det_ctx->events++;
 }
 
+bool DetectMd5ValidateCallback(
+        const Signature *s, const DetectContentData *cd, const char **sigerror)
+{
+    if (cd->flags & DETECT_CONTENT_NOCASE) {
+        *sigerror = "md5-like keyword should not be used together with "
```

Comment:
having a single function, we lose the precision of the keyword in the constant `sigerror`

I guess we should improve the warning log to use the current buffer name...